pywhat/Data/regex.json

```diff
@@ -394,6 +394,18 @@
           "Networking"
        ]
     },
+    {
+      "Name": "Amazon Standard Identification Number (ASIN)",
+      "Regex": "(?:[/dp/]|$)([A-Z0-9]{10})",

```

Comment:
We want all regexes to have hardcoded boundaries (^$.) Can you add these? Btw, I see that some entries do not have those either. It would be fantastic if you could add boundaries.

Reply:
I'm not super confident with regex just yet. Is there a line that can serve as an ideal example of this?

Reply:
I can explain it to you. For example, you want to match to the word `amazon`. By default regex should only match if `amazon` is in its own line. It should match to this line:
`amazon`
And not to `amazon` in this line:
`amazon test`

To accomplish this you need to add `^` at the beginning of regex which means start of line, and `$` at the end of regex which means the end of line.

So change your regex to this: `^(your regex)$`. Some others like SSH and API ones that were added recently also need this. It would be amazing if you could fix that. 😁

You can experiment with regex on a website like `regex101.com`. If you have any further questions, I am happy to help. 😄

Reply:
Thanks for the detailed explanation! I'll be sure to reach out with any questions.